test/compilable/issue21203.d

```diff
@@ -0,0 +1,144 @@
+
+template ScopeClass(C , string name = C.stringof)
+//if (is(C == class) && __traits(getLinkage, C) == "C++")
+{
+    //enum name = C.stringof;
+    enum ns = __traits(getCppNamespaces,C);
+    extern(C++, class)
+    {
+        extern(C++,(ns))
+        {
+            pragma(mangle, name, C)
```

Comment:
I also don't understand why `C` is given here? I must be overlooking something obvious :/